pgmpy/factors/FactorSet.py

```diff
@@ -0,0 +1,191 @@
+#!/usr/bin/env python3
+from pgmpy.factors import Factor
+from functools import reduce
+
+
+class FactorSet:
```

Comment:
@abinashpanda I think we should also have `add` and `remove` methods to add/remove factors from `FactorSet`? What do you think?